src/python/pants/init/plugin_resolver.py

```diff
@@ -3,36 +3,151 @@
 
 from __future__ import annotations
 
+import json
 import logging
+import os
+import shlex
 import site
 import sys
-from collections.abc import Iterable
+from collections.abc import Iterable, Mapping
 from dataclasses import dataclass
+from pathlib import PurePath
+from textwrap import dedent  # noqa: PNT20
 from typing import cast
 
 from pkg_resources import Requirement, WorkingSet
 from pkg_resources import working_set as global_working_set
 
+from pants.backend.python.subsystems.python_native_code import PythonNativeCodeSubsystem
+from pants.backend.python.subsystems.repos import PythonRepos
+from pants.backend.python.subsystems.setup import PythonSetup
 from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
-from pants.backend.python.util_rules.pex import PexRequest, VenvPex, VenvPexProcess
+from pants.backend.python.util_rules.pex_cli_tool import PexPEX
+from pants.backend.python.util_rules.pex_cli_tool import rules as pex_cli_tool_rules
 from pants.backend.python.util_rules.pex_environment import PythonExecutable
-from pants.backend.python.util_rules.pex_requirements import PexRequirements
+from pants.core.subsystems.python_bootstrap import PythonBootstrap
+from pants.core.util_rules.adhoc_binaries import PythonBuildStandaloneBinary
 from pants.core.util_rules.environments import determine_bootstrap_environment
+from pants.core.util_rules.system_binaries import BashBinary
 from pants.engine.collection import DeduplicatedCollection
-from pants.engine.env_vars import CompleteEnvironmentVars
+from pants.engine.env_vars import CompleteEnvironmentVars, EnvironmentVars, EnvironmentVarsRequest
 from pants.engine.environment import EnvironmentName
+from pants.engine.fs import CreateDigest, Digest, Directory, FileContent, MergeDigests
 from pants.engine.internals.selectors import Params
 from pants.engine.internals.session import SessionValues
-from pants.engine.process import ProcessCacheScope, ProcessResult
+from pants.engine.process import Process, ProcessCacheScope, ProcessResult
 from pants.engine.rules import Get, QueryRule, collect_rules, rule
 from pants.init.bootstrap_scheduler import BootstrapScheduler
 from pants.option.global_options import GlobalOptions
 from pants.option.options_bootstrapper import OptionsBootstrapper
+from pants.util.frozendict import FrozenDict
 from pants.util.logging import LogLevel
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True)
```

Comment:
Vendored these two classes and function from `VenvScriptWriter` to avoid the dependency on Python backend.